src/useTimer.ts

```diff
@@ -19,92 +19,58 @@ export interface IValues {
   time: number;
 }
 
-const initialConfig: IConfig = {
-  endTime: null,
-  initialTime: 0,
-  interval: 1000,
-  step: 1,
-  timerType: 'INCREMENTAL',
-};
-
-export const useTimer = (config?: Partial<IConfig>): IValues => {
-  const { endTime, initialTime, interval, onTimeOver, step, timerType } = {
-    ...initialConfig,
-    ...config,
-  };
-  const intervalRef = useRef<NodeJS.Timeout | null>(null);
-  const pausedTimeRef = useRef<number | null>(null);
-  const [shouldResetTime, setShouldResetTime] = useState(false);
+export const useTimer = ({
+  initialTime = 0,
+  interval = 1000,
+  step = 1,
+  timerType = 'INCREMENTAL',
+  endTime,
+  onTimeOver,
+}: Partial<IConfig> = {}): IValues => {
   const [time, setTime] = useState(initialTime);
   const [isRunning, setIsRunning] = useState(false);
+  const [isCountFinished, setIsCountFinished] = useState(false);
 
-  const cancelInterval = () => {
-    if (intervalRef.current) {
-      clearInterval(intervalRef.current);
-      intervalRef.current = null;
-      setIsRunning(false);
-    }
-  };
-
-  const createInterval = () => {
-    setIsRunning(true);
-
-    intervalRef.current = setInterval(() => {
-      setTime(previousTime => {
-        const newTime =
-          timerType === 'INCREMENTAL'
-            ? previousTime + step
-            : previousTime - step;
-
-        if (endTime !== null && newTime === endTime) {
-          stopTimerWhenTimeIsOver();
-        }
-
-        return newTime;
-      });
-    }, interval);
-  };
-
-  const pause = () => {
-    pausedTimeRef.current = time;
-
-    cancelInterval();
-  };
-
-  const reset = () => {
-    pausedTimeRef.current = null;
-
-    cancelInterval();
-    resetTime();
-  };
-
-  const resetTime = () => {
+  const reset = useCallback(() => {
```

Comment:
I don't agree with the use of **useCallback**: https://github.com/thibaultboursier/use-timer/pull/10#pullrequestreview-408139335